src/ionosphere/waccmx/edyn_esmf.F90

```diff
@@ -389,30 +293,42 @@ subroutine edyn_esmf_init( mpi_comm )
       integer              :: petCount
       integer              :: iam
       integer              :: npes
-      integer, allocatable :: petlist(:)
+      integer              :: mag_npes !!XXgoldyXX: get this from mag grid?
+      type(ESMF_DELayout)  :: geo_delayout
+      type(ESMF_DELayout)  :: mag_delayout
       type(ESMF_VM)        :: vm_init
-      type(ESMF_VM)        :: vm_curr
-
-      type(ESMF_GridComp)  :: phys_comp ! physics grid mesh
+#endif
+!!XXgoldyXX: ^ remove?
+      character(len=*), parameter :: subname = 'edyn_esmf_init'
 
+!!XXgoldyXX: v remove?
+#if 0
+      !! Create distribution and PE information for each grid
       call ESMF_VMGetCurrent(vm_init, rc=rc)
-      call edyn_esmf_chkerr('ESMF_VMGetCurrent', rc)
+      call edyn_esmf_chkerr(subname, 'ESMF_VMGetCurrent', rc)
       call ESMF_VMGet(vm_init, localPet=localPet, petCount=petCount)
-      call edyn_esmf_chkerr('ESMF_VMGet', rc)
+      call edyn_esmf_chkerr(subname, 'ESMF_VMGet', rc)
       call mpi_comm_size(mpi_comm, npes, rc)
       call mpi_comm_rank(mpi_comm, iam, rc)
-
-      ! Collect all the PETS for each instance
+      ! Collect all the PETS for each instance for phys grid
       allocate(petlist(npes))
       call mpi_allgather(localPet, 1, MPI_INTEGER, petlist, 1, &
            MPI_INTEGER, mpi_comm, rc)
-      ! Now, we should be able to create a gridded component with these pets
-      phys_comp = ESMF_GridCompCreate(petList=petList, name=trim(inst_name), rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompCreate '//trim(inst_name), rc)
-      call ESMF_GridCompSetServices(phys_comp, edyn_gcomp_SetServices, rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompSetServices '//trim(inst_name), rc)
-      call ESMF_GridCompInitialize(phys_comp, rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompRun '//trim(inst_name), rc)
+      phys_delayout = ESMF_DELayoutCreate(deCount=npes, petList=petlist, rc=rc)
+      call edyn_esmf_chkerr(subname, 'phys ESMF_DELayoutCreate', rc)
+      geo_delayout = ESMF_DELayoutCreate(deCount=geo_npes,                    &
+           petList=petlist(1:geo_npes), rc=rc)
+      call edyn_esmf_chkerr(subname, 'geo ESMF_DELayoutCreate', rc)
+      geo_distgrid = ESMF_DistGridCreate(minIndex=(/1/),                      &
+           maxIndex=(/geo_npes/), delayout=geo_delayout, rc=rc)
+      call edyn_esmf_chkerr(subname, 'geo ESMF_DistGridCreate', rc)
+      mag_npes = npes !!XXgoldyXX: get this from mag grid?
+      mag_delayout = ESMF_DELayoutCreate(deCount=mag_npes,                    &
+           petList=petlist(1:mag_npes), rc=rc)
+      call edyn_esmf_chkerr(subname, 'mag ESMF_DELayoutCreate', rc)
+      mag_distgrid = ESMF_DistGridCreate(minIndex=(/1/),                      &
+           maxIndex=(/mag_npes/), delayout=mag_delayout, rc=rc)
+      call edyn_esmf_chkerr(subname, 'mag ESMF_DistGridCreate', rc)
```

Comment:
Can geo_distgrid and mag_distgrid be disjointed from phys_mesh?  That is can the ionosphere grids be decomposed across a set of PETs which is a subset of the phys_mesh PETs?

Reply:
These are not used (partially for the reasons you cite, partially because I did not need them). I think they are `if 0`ed out except for the declarations in the module data.
The decomposition of the grids does not matter as long as the grid is created in the same VM as the mesh (as they are in the PR code). Some PEs will just not have any columns assigned for the geo and mag grids.

Reply:
Interesting.  So maybe we can run edynamo and oplus on a subset of tasks like we do for FV dycore.